amida-tech/indaba-client
# Task/inba 569 metadata survey

#### What does this PR do?
This PR also satisfies INBA-568. 
Please review https://github.com/amida-tech/indaba-client/pull/301 and https://github.com/amida-tech/indaba-client/pull/303 before this one. It will need some merges.

Adds meta data support for links for questions. Also removes the links/file fields during the survey builder.

**Note:** Yes, I am updating the reducer data based on success of the call rather than just calling the survey for answers again. This will go away once I have time to ~~set the code on fire and dance a jig~~ update to redux-forms. 

**Note 2:** An update for how survey services handles meta data is coming soon after this per Chris Porter's changes.

**Note 3:** There's a possibility the entire survey microservice may get a total rewrite.

**Note 4:** James is a victim of Tzeentch, the Raven God and Changer of Ways. He may not be entirely sane right this minute. 

#### Related JIRA tickets:
[INBA-569](https://jira.amida-tech.com/browse/INBA-569)
SER-144.

#### How should this be manually tested?
Create a new project with everything. In the survey builder, add some questions (recommend: long text, number, multiple choice as working question types).  For a few of the questions, be sure to click the "Add a Link" and "Attach File" checkboxes.

Ensure that you do not see the fields for these features display here.

Finish and assign some users to tasks. Log out and log in as one of these users. Go to the Task itself and click the questions. Fill out an answer a question. 

Ensure that the user sees the meta data fields (link, file).

Click the "Add File" button. Ensure that you get a toast pop up ensuring the user that the feature is "coming soon."

Add publication data for link, title, author and date. Ensure that the data persists after refreshing the page.

#### Background/Context
**Update your survey microservice. Update your survey microservice. Update your survey microservice.**
... before you ask.

#### Screenshots (if appropriate):
